src/main/java/at/aau/ase/cl/domain/AuthorEntity.java

```diff
@@ -0,0 +1,61 @@
+package at.aau.ase.cl.domain;
+
+import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Id;
+import jakarta.persistence.Index;
+import jakarta.persistence.Table;
+
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Entity
+@Table(name = "book_author", indexes = {
+        @Index(name = "idx_book_author_key", columnList = "key", unique = true)
+})
+public class AuthorEntity extends PanacheEntityBase {
+    @Id
+    public UUID id;
+
+    @Column(nullable = false)
+    public String key;
+
+    @Column(nullable = false)
+    public String name;
+
+    public static AuthorEntity findByKey(String key) {
+        return find("key", key).firstResult();
+    }
+
+    /**
+     * Computes the key for this author entity based on the name.
+     * @return the computed key
+     */
+    public String computeKey() {
```

Comment:
Good method for generating Key. If I understand correctly in the openlibrary author_name is "J.R.R. Tolkien" and the name is in alternative names (see image). There is also a key coming from the API (author_key) which is not used by our application. This could be confusing but is fine, please add your definition of the author key (how it is generated) to [our domain diagram](https://github.com/AAU-ASE24W-group-E/crowd-library/blame/main/docs/domain-analysis/domain-diagram.md).
![grafik](https://github.com/user-attachments/assets/e116fde8-7b35-4c5a-9104-98c6ef54f208)

Reply:
explanation added